### PR TITLE
refactor mixin

### PR DESCRIPTION
**Pull request by knaaptime**
merge this to get the tests to run upstream

### CODE REVIEW

File: libpysal/weights/tests/test_contiguity.py [hunk 1]
```diff
@@ -109,6 +117,8 @@ def test_from_geodataframe_order(self):
             assert w.id_order[:5] == expected
 
     def test_from_xarray(self):
+        _  = pytest.importorskip("xarray")
```
Comment (jGaboardi) [inline libpysal/weights/tests/test_contiguity.py]: From what I understand here don't need to actually declare anything, we can have `pytest.importorskip("xarray")`

File: libpysal/weights/tests/test_raster.py [hunk 2]
```diff
@@ -2,13 +2,12 @@
...
 import pandas as pd
 import pytest
-from xarray import DataArray
-
 from .. import raster
 
 
 class Testraster:
     def setup_method(self):
+        xarray = pytest.importorskip("xarray")
```
Comment (jGaboardi) [inline libpysal/weights/tests/test_raster.py]: Same for here.

File: libpysal/weights/tests/test_raster.py [hunk 3]
```diff
@@ -77,9 +76,10 @@ def test_da2_wsp(self):
         assert w2.index.tolist() == self.da2.to_series().index.tolist()
 
     def test_w2da(self):
+        xarray = pytest.importorskip("xarray")
```
Comment (jGaboardi) [inline libpysal/weights/tests/test_raster.py:79]: And here.

File: libpysal/weights/tests/test_raster.py (hunk 2 above)
Comment (knaaptime) [reply libpysal/weights/tests/test_raster.py]: no we use xarray below

File: libpysal/weights/tests/test_raster.py (hunk 3 above)
Comment (knaaptime) [reply libpysal/weights/tests/test_raster.py:79]: need it here too

**Comment by jGaboardi** [comment]
Wait this is in my fork's `main`.

**Comment by knaaptime** [comment]
no its against your branch

Screenshot 2023-11-04 at 5 05 28 PM

**Comment by jGaboardi** [comment]
Need to be in [jGaboardi:remove_xarray_hard_dep](https://github.com/jGaboardi/libpysal/tree/remove_xarray_hard_dep)

**Comment by knaaptime** [comment]
>Need to be in [jGaboardi:remove_xarray_hard_dep](https://github.com/jGaboardi/libpysal/tree/remove_xarray_hard_dep?rgh-link-date=2023-11-05T00%3A06%3A08Z)

it is

**Comment by jGaboardi** [comment]
hmm., yeah, but it's not updating against https://github.com/pysal/libpysal/pull/629... Not sure why.

**Comment by knaaptime** [comment]
>hmm., yeah, but it's not updating against https://github.com/pysal/libpysal/pull/629... Not sure why.

because you need to merge it here first

**Comment by knaaptime** [comment]
i think you need to hit the workflow approval

**Comment by knaaptime** [comment]
but you wont see these changes in the upstream pr until you merge here

**Comment by knaaptime** [comment]
(usually when martin does it, he commits directly to my branch, so it shows up immediately in the upstream PR. Here, i'm asking you to merge my changes into your branch, which will then be included in your PR upstream)

**Pull request merged**